src/common/transformations/src/transformations/fp16_compression/mark_decompression_convert_constant_folding.cpp

```diff
@@ -68,3 +68,28 @@ pass::KeepConstAndDecompression::KeepConstAndDecompression() {
     auto m = std::make_shared<pattern::Matcher>(node_pattern, matcher_name);
     register_matcher(m, callback);
 }
+
+pass::KeepConstAndDecompressionForMatMul::KeepConstAndDecompressionForMatMul() {
+    MATCHER_SCOPE(KeepConstAndDecompressionForMatMul);
+    auto matmul = pass::pattern::wrap_type<opset8::MatMul>();
+
+    matcher_pass_callback callback = [=](pass::pattern::Matcher& m) {
+        auto node = m.get_match_root();
+
+        // input to matmul is decompression Convert
+        const auto& inp_convert = node->input_value(1).get_node_shared_ptr();
+        if (!is_type<opset8::Convert>(inp_convert) || !is_decompression(inp_convert))
+            return false;
+
+        disable_constant_folding(inp_convert);
+
+        if (!is_type<opset8::Constant>(inp_convert->input_value(0).get_node_shared_ptr()))
+            return true;
```

Comment:
I think it should be 
"return false"
`true` means, that somethings has changed in the graph (nodes replaces, edges changed) and we need to run  additional shape re-calculation and validation

Reply:
changed to `false` here and in `KeepConstAndDecompression()` above, thanks